modules/@apostrophecms/schema/index.js

```diff
@@ -595,6 +588,45 @@ module.exports = {
       name: 'group' // visual grouping only
     });
 
+    self.addFieldType({
+      name: 'range',
+      vueComponent: 'AposInputRange',
+      convert: async function (req, field, data, object) {
+        object[field.name] = self.apos.launder.float(data[field.name], field.def, field.min, field.max);
+        if (field.required && (_.isUndefined(data[field.name]) || !data[field.name].toString().length)) {
+          throw self.apos.error('required');
+        }
+        if (data[field.name] && isNaN(parseFloat(data[field.name]))) {
+          throw self.apos.error('invalid');
+        }
+        // Allow for ranges to go unset
+        // `min` here does not imply requirement, it is the minimum value the range UI will represent
+        if (!data[field.name] || data[field.name] < parseFloat(field.min)) {
+          object[field.name] = null;
+        }
+      },
+      validate: function (field, options, warn, fail) {
+        if (!field.min) {
+          fail('Property "min" must be set.');
+        }
+        if (!field.max) {
+          fail('Property "max" must be set.');
+        }
+        if (isNaN(parseFloat(field.max))) {
+          fail('Property "max" must be an integer.');
+        }
+        if (isNaN(parseFloat(field.min))) {
```

Comment:
The message is that it must be an integer, but this (and above) is testing for float.

Reply:
HTML range inputs aren't limited to integers, so I don't know why ours should be.

Reply:
`parseFloat` won't error if you give it a stringified float .. you want to change the error language to something akin to that? (sounds wordy which is why i didnt bother)

Reply:
I'm saying if we're requiring a float, the message should say that it must be a float value. This error is going to the dev, who can/should know the difference.